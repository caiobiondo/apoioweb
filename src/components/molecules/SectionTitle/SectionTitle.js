import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'natura-ui';
import IconButton from 'material-ui/IconButton';
import { FormattedMessage } from 'react-intl';

import {
  SectionTitleWrapper,
  Title,
  ExpandableIconWrapper,
  ExpandableIconButton,
  ExpandableIcon,
} from './SectionTitle.styles';

const SectionTitle = props => {
  const { iconName, children, value, expandable, expanded, onToggle } = props;

  return (
    <SectionTitleWrapper {...props}>
      {iconName && <Icon file={iconName} />}
      {value ? (
        <Title>
          <FormattedMessage id={value} />
        </Title>
      ) : (
        children
      )}
      {expandable && (
        <ExpandableIconWrapper>
          <IconButton onClick={onToggle} style={ExpandableIconButton} iconStyle={ExpandableIcon}>
            {expanded ? <Icon file="ico_arrow_down" /> : <Icon file="ico_arrow_right" />}
          </IconButton>
        </ExpandableIconWrapper>
      )}
    </SectionTitleWrapper>
  );
};

SectionTitle.propTypes = {
  iconName: PropTypes.string,
  value: PropTypes.string,
};

export default SectionTitle;
