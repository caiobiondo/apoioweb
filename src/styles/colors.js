//
// Color Specs
// -------------------------------------
export const black = '#000';
export const white = '#fff';

export const gray100 = '#eeeeee';
export const gray150 = '#dddddd';
export const gray200 = '#bbbbbb';
export const gray300 = '#8e8e8e';
export const gray600 = '#606060';
export const gray890 = '#333333';
export const gray900 = '#222222';

export const orange100 = '#f4a652';
