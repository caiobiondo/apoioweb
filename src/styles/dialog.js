export const dialogContainer = {
  maxWidth: 'none',
};

export const dialogContent = {
  fontFamily: 'Roboto',
  padding: '12px 25px',
};

export const dialogTitle = {
  fontWeight: 800,
};

export const dialogActions = {
  fontFamily: 'Roboto',
};
