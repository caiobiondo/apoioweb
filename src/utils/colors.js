export const CARD_COLOR = {
  SUCCESS: 'SUCCESS',
  WARNING: 'WARNING',
  DANGER: 'DANGER',
  INFO: 'INFO'
};

export function getGradientColor(color) {
  if (color === CARD_COLOR.SUCCESS) {
    return 'linear-gradient(#89ddd2, #cddb50)';
  }
  if (color === CARD_COLOR.WARNING) {
    return 'linear-gradient(#fcdf4f, #d5bf3e)';
  }
  if (color === CARD_COLOR.DANGER) {
    return 'linear-gradient(#f19632, #de3b2f)';
  }
  if (color === CARD_COLOR.INFO) {
    return 'linear-gradient(#9271cb, #6091ca)';
  }

  return 'transparent';
}

export function getSolidColor(color) {
  if (color === CARD_COLOR.SUCCESS) {
    return '#bed164';
  }
  if (color === CARD_COLOR.WARNING) {
    return '#dac75e';
  }
  if (color === CARD_COLOR.DANGER) {
    return '#e25e57';
  }
  if (color === CARD_COLOR.INFO) {
    return '#7ea7be';
  }

  return '#404040';
}
