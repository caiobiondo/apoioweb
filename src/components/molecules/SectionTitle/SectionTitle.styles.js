import styled from 'styled-components';
import { fs18 } from 'styles/typography';
import { gray890, orange100 } from 'styles/colors';

export const SectionTitleWrapper = styled.div`
  display: flex;

  align-items: center;
  vertical-align: middle;
  font-size: ${fs18};
  color: ${props => props.color || gray890};
  margin-bottom: ${props => {
    return props.expanded === false ? 0 : '30px';
  }};
  position: relative;

  svg {
    margin-right: 20px;
    display: inline-block;

    fill: ${props => props.svgFill || orange100};
    height: 20px;
    width: 20px;
  }
`;

export const Title = styled.div`
  flex: 1;
`;

export const ExpandableIconWrapper = styled.div`
  svg {
    margin-right: 0;
  }
`;

export const ExpandableIconButton = {
  width: 36,
  height: 36,
  padding: 0,
};

export const ExpandableIcon = {
  width: 72,
  height: 72,
  padding: 16,
};
