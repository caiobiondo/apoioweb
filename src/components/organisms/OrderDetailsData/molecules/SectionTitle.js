import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'natura-ui';
import { SectionTitleWrapper } from './SectionTitle.styles';

const SectionTitle = props => {
  const { iconName, children } = props;

  return (
    <SectionTitleWrapper {...props}>
      <Icon file={iconName} />
      {children}
    </SectionTitleWrapper>
  );
};

SectionTitle.propTypes = {
  iconName: PropTypes.string,
};

export default SectionTitle;
