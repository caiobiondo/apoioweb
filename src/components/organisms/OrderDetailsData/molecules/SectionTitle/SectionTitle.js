import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'natura-ui';
import { SectionTitleWrapper } from './SectionTitle.styles';
import { FormattedMessage } from 'react-intl';

const SectionTitle = props => {
  const { iconName, children, value } = props;

  return (
    <SectionTitleWrapper {...props}>
      <Icon file={iconName} />
      {value ? <FormattedMessage id={value} /> : children}
    </SectionTitleWrapper>
  );
};

SectionTitle.propTypes = {
  iconName: PropTypes.string,
};

export default SectionTitle;
