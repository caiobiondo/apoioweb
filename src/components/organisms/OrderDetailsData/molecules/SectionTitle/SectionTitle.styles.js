import styled from 'styled-components';
import { fs18 } from 'styles/typography';
import { gray890, orange100 } from 'styles/colors';

export const SectionTitleWrapper = styled.div`
  display: flex;

  vertical-align: middle;
  font-size: ${fs18};
  color: ${props => props.color || gray890};
  margin-bottom: 30px;

  svg {
    margin-right: 20px;
    display: inline-block;

    fill: ${props => props.svgFill || orange100};
    height: 20px;
    width: 20px;
  }
`;
