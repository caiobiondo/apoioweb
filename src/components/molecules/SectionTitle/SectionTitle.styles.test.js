import React from 'react';
import { shallow } from 'enzyme';
import { SectionTitleWrapper } from './SectionTitle.styles';
import { gray890 } from 'styles/colors';
import 'jest-styled-components';

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides);

  const result = shallow(<SectionTitleWrapper {...props} />);

  return {
    props,
    result,
  };
};

describe('SectionTitle styles', () => {
  it('renders SectionTitleWrapper', () => {
    // given
    // when
    const { result } = setup({});

    // then
    expect(result).toMatchSnapshot();
  });

  it('renders SectionTitle with default values', () => {
    // given
    // when
    const { result } = setup({});

    // then
    expect(result).toMatchSnapshot();
    expect(result).toHaveStyleRule('color', gray890);
  });

  it('renders SectionTitle with selected color', () => {
    // given
    const color = '#000';

    // when
    const { result } = setup({ color });

    // then
    expect(result).toHaveStyleRule('color', color);
  });

  it('renders SectionTitle with default svg color', () => {
    // given
    const svgFill = '#fff';

    // when
    const { result } = setup({ svgFill });

    // then
    expect(result).toMatchSnapshot();
  });
});
