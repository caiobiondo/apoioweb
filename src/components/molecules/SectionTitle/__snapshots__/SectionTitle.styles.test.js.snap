// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle styles renders SectionTitle with default svg color 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  font-size: 18px;
  color: #333333;
  margin-bottom: 30px;
  position: relative;
}

.c0 svg {
  margin-right: 20px;
  display: inline-block;
  fill: #fff;
  height: 20px;
  width: 20px;
}

<div
  className="c0"
/>
`;

exports[`SectionTitle styles renders SectionTitle with default values 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  font-size: 18px;
  color: #333333;
  margin-bottom: 30px;
  position: relative;
}

.c0 svg {
  margin-right: 20px;
  display: inline-block;
  fill: #f3971f;
  height: 20px;
  width: 20px;
}

<div
  className="c0"
/>
`;

exports[`SectionTitle styles renders SectionTitleWrapper 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  font-size: 18px;
  color: #333333;
  margin-bottom: 30px;
  position: relative;
}

.c0 svg {
  margin-right: 20px;
  display: inline-block;
  fill: #f3971f;
  height: 20px;
  width: 20px;
}

<div
  className="c0"
/>
`;
